### PR TITLE
refactor: Operators Simplifications

### PR DESCRIPTION
**Pull request by maximebrugel**
### Simplifications 
- The operators used `create2` to retrieve the storage address. Therefore, it needed to `self` parameter given by the `NestedFactory`.  Now instead, we are storing the storage address as an `immutable` variable to avoid the `create2` mechanism.
- Remove `commit`/`revert` selector choice. Now the `OperatorResolver` will store the `address` and `bytes4` selector. Then, the caller can choice a whitelisted "operator + selector" and not only the "operator + commit or revert selector" (via the `_commit` flag). The operator could have more than two functions and the call is more gas efficient.

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -77,9 +77,7 @@ When we want to "reverse" an action that we have "committed", we will "**revert*
 Since the operators are called via `delegatecall`: _how can we store/retrieve useful data?_
 <br>In fact, we cannot trust the Factory to provide all the data, like the address of the protocol. It must be stored and managed by the owner.
 
-When deploying an operator, it will also deploy the storage contract using `CREATE2` and transfer the ownership to `msgSender()`.
-
-This way, the operator can retrieve the storage address (without storing it) and get the data.
+When deploying an operator, it will also deploy the storage contract and transfer the ownership to `msgSender()`.
```
Comment (maximebrugel) [inline README.MD:80]: Edit "One language" section to remove the mention of "commit/revert"

File: contracts/MixinOperatorResolver.sol [hunk 2]
```diff
@@ -63,9 +69,39 @@ abstract contract MixinOperatorResolver {
...
+        // We are concatenating the selector + given params
+        bytes memory data;
+        (success, data) = _operator.implementation.delegatecall(
+            bytes.concat(
+                _operator.selector,
+                _order.callData
+            )
+        );
+
+        if (success) {
```
Comment (maximebrugel) [inline contracts/MixinOperatorResolver.sol:100]: Nice!

File: README.MD (hunk 1 above)
[the remark this answers is not in the document]
Comment (maximebrugel) [reply README.MD:84]: Done

File: contracts/OperatorResolver.sol [hunk 3]
```diff
@@ -9,17 +9,17 @@ import "@openzeppelin/contracts/access/Ownable.sol";
...
-    function getAddress(bytes32 name) external view override returns (address) {
+    function getAddress(bytes32 name) external view override returns (Operator memory) {
         return operators[name];
     }
 
     /// @inheritdoc IOperatorResolver
-    function requireAndGetAddress(bytes32 name, string calldata reason) external view override returns (address) {
-        address _foundAddress = operators[name];
-        require(_foundAddress != address(0), reason);
+    function requireAndGetAddress(bytes32 name, string calldata reason) external view override returns (Operator memory) {
```
[the remark this answers is not in the document]
Comment (adrien-supizet) [reply contracts/OperatorResolver.sol:20]: I also think we should rename the functions as suggested 👍

File: scripts/deployAll.ts [hunk 4]
```diff
@@ -72,7 +70,7 @@ async function main(): Promise<void> {
     console.log("ZeroExOperator deployed : ", zeroExOperator.address);
 
     // Add ZeroExStorage address
-    deployments.push({name: "ZeroExStorage", address: await zeroExOperator.storageAddress(zeroExOperator.address)})
+    deployments.push({ name: "ZeroExStorage", address: await zeroExOperator.storageAddress(zeroExOperator.address) })
```
[the remark this answers is not in the document]
Comment (maximebrugel) [reply scripts/deployAll.ts:73]: I was wrong... Just saw this line `let iface = new ethers.utils.Interface([signature]);`

**Pull request merged**